src/appmixer/google/gmail/CreateDraft/CreateDraft.js

```diff
@@ -22,15 +23,36 @@ module.exports = {
             mail.from = `${mail.sender} <${context.profileInfo.email}>`;
         }
 
+        if (context.messages.in.content.cc) {
+            mail.cc = context.messages.in.content.cc;
+        }
+
+        if (context.messages.in.content.bcc) {
+            mail.bcc = context.messages.in.content.bcc;
+        }
+
+        const { attachments = {} } = context.messages.in.content;
+        const fileIds = (attachments.ADD || [])
+            .map(attachment => (attachment.fileId || null))
+            .filter(fileId => fileId !== null);
+
+        mail.attachments = await Promise.map(fileIds, async (fileId) => {
+            const fileInfo = await context.getFileInfo(fileId);
+            const fileStream = await context.getFileReadStream(fileId);
+            return { filename: fileInfo.filename, content: fileStream };
+        });
+
+        console.log('Mail object before building:', mail); // Debugging
+
         return new Promise((resolve, reject) => {
-            mailcomposer(mail).build((err, email) => {
+            mailcomposer({ ...mail, keepBcc: true }).build((err, email) => {  // Added keepBcc option
```

Comment:
@vtalas somehow bcc property is not being retained in both cases SendEmail and CreateDraft, Here in this line I have added this property keepBcc after looking here [https://nodemailer.com/extras/mailcomposer/#bcc](https://nodemailer.com/extras/mailcomposer/#bcc) I will try to make it work